### PR TITLE
fix(hz): wrong package name under windows

### PR DESCRIPTION
**Pull request by FGYFFFF**
#### What type of PR is this?

fix
#### Check the PR title.

- [ ] This PR title match the format: \<type\>(optional scope): \<description\>
- [ ] The description of this PR title is user-oriented and clear enough for others to understand.

#### (Optional) Translate the PR title into Chinese.

修复 windows 下 package 的 error
#### (Optional) More detail description for this PR(en: English/zh: Chinese).

en: Fix package error on windows
zh(optional): 修复 windows 下 package 的 error

#### Which issue(s) this PR fixes:

https://github.com/cloudwego/hertz/issues/642

### CODE REVIEW

File: cmd/hz/config/argument.go [hunk 1]
```diff
@@ -228,12 +228,12 @@ func (arg *Argument) checkPackage() error {
...
 				return fmt.Errorf("can not get relative path, err :%v", err)
 			}
-			module = filepath.Join(module, rel)
-			arg.Gomod = module
+			arg.Gomod = filepath.Join(module, rel)
 		}
 		if len(arg.Gomod) == 0 {
 			arg.Gomod = arg.Gopkg
 		}
+		arg.Gomod = util.PathToImport(module, "")
```
Comment (welkeyever) [inline cmd/hz/config/argument.go:236]: Seems that line 233~235 become meaningless according to this change?

Comment (FGYFFFF) [reply cmd/hz/config/argument.go]: done

File: cmd/hz/util/env.go [hunk 2]
```diff
@@ -103,6 +103,10 @@ func SearchGoMod(cwd string, recurse bool) (moduleName, path string, found bool)
 			break
 		}
 		cwd = filepath.Dir(cwd)
+		// "C:\" will return "C:\" under windows
+		if cwd == filepath.Dir(cwd) {
```
Comment (welkeyever) [inline cmd/hz/util/env.go:107]: What is the purpose of this check? Maybe adding more description in comment?

Comment (FGYFFFF) [reply cmd/hz/util/env.go:107]: Not determining the root directory under windows， it can lead to a dead loop

**Review by Duslia — DISMISSED** [review]

**Review by Duslia — APPROVED** [review]

**Review by welkeyever — APPROVED** [review]

**Pull request merged**